### PR TITLE
feat: adding retries to Stages (tekton based solution)

### PR DESCRIPTION
**Pull request by joseblas**
#### Submitter checklist

- [X] Change is code complete and matches issue description.
- [X] Change is covered by existing or new tests.

#### Description
  Since Tekton already provides retries, one new param has to be passed to it.
 This PR just makes sure that the param coming from jenkins-x.yml goes into Tekton Pipeline CRD.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/jx/cmd/test_data/step_create_task/command-as-multiline-script/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: tekton.dev/v1alpha1
```
Comment (abayer) [inline pkg/jx/cmd/test_data/step_create_task/command-as-multiline-script/pipeline.yml]: This file should be removed - I'm guessing it's a lurking artifact from before the `cmd` package got reorganized. =)

File: pkg/tekton/syntax/pipeline.go [hunk 2]
```diff
@@ -1534,6 +1533,13 @@ func (j *ParsedPipeline) GenerateCRDs(pipelineIdentifier string, buildIdentifier
 			return nil, nil, nil, err
 		}
 
+		o := stage.Stage.Options
+		if o.RootOptions != nil {
+			if o.Retry > 0 {
+				stage.Stage.Options.Retry = s.Options.Retry
+				log.Infof("Generating CRDs with %d retries", stage.Stage.Options.Retry)
```
Comment (abayer) [inline pkg/tekton/syntax/pipeline.go]: Do we really want to be logging here? I tend to think we don't.

Comment (joseblas) [reply pkg/tekton/syntax/pipeline.go]: Well, I think logging is good in general; is good for analysing possible bug or weird situations

File: pkg/jx/cmd/test_data/step_create_task/command-as-multiline-script/pipeline.yml (hunk 1 above)
Comment (abayer) [reply pkg/jx/cmd/test_data/step_create_task/command-as-multiline-script/pipeline.yml]: But it's not getting used in the tests - the one in `pkg/jx/cmd/step/create/test_data` is.

File: pkg/tekton/syntax/pipeline.go (hunk 2 above)
Comment (abayer) [reply pkg/tekton/syntax/pipeline.go]: Then a better logging message makes sense - like, say "setting retries to %d for stage %s".

**Pull request merged**